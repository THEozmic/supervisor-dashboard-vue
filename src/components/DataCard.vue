<template>
  <div :class="`card card--${color}`">
    <div class="card__title">{{ title }}</div>
    <div class="card__content">{{ content }}</div>
  </div>
</template>

<script>
export default {
  name: "DataCard",
  props: {
    title: {
      type: String,
      required: true
    },
    content: {
      type: String,
      required: true
    },
    color: {
      type: String,
      required: true
    }
  }
};
</script>

<style scoped>
.card {
  display: inline-block;
  padding: 20px;
  margin-bottom: 30px;
  color: #ffffff;
  width: 30%;
  box-shadow: 0 15px 35px 0 rgba(42, 51, 83, 0.12),
    0 5px 15px rgba(0, 0, 0, 0.06);
  border-radius: 4px;
}

.card--blue {
  background-color: #375bae;
}

.card--green {
  background-color: #64ae37;
}

.card--red {
  background-color: #ae377f;
}

.card__title {
  font-size: 30px;
}

.card__content {
  font-size: 20px;
}

@media screen and (max-width: 999px) {
  .card {
    width: 100%;
  }
}
</style>
