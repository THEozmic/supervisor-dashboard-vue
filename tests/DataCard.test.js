import { shallowMount } from '@vue/test-utils'
import DataCard from '@/components/DataCard.vue'

describe('DataCard.vue', () => {

  it('applies correct color class', () => {
    const wrapper = shallowMount(DataCard, {
      propsData: { title: 'Test title', content: 'Test content', color: 'blue' }
    })
    expect(wrapper.find('.card').classes()).toContain('card--blue')
  })

  it('matches snapshot', () => {
    const wrapper = shallowMount(DataCard, {
      propsData: { title: 'Test title', content: 'Test content', color: 'blue' }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
