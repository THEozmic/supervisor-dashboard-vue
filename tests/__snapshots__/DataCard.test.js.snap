// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataCard.vue matches snapshot 1`] = `
<div class="card card--blue">
  <div class="card__title">Test title</div>
  <div class="card__content">Test content</div>
</div>
`;
